src/components/capture/create/index.tsx

```diff
@@ -135,130 +134,108 @@ function CaptureCreation() {
         selectors.form.exitWhenLogsClose
     );
 
-    // Local state
-    const [catalogResponse, setCatalogResponse] = useState<any | null>(null);
-
-    const cleanUpEditor = () => {
-        if (Object.keys(resourcesFromEditor).length > 0) {
-            clearResourcesFromEditor();
-        }
-    };
+    //Editor state
+    const draftId = useEditorStore(editorStoreSelectors.draftId);
+    const setDraftId = useEditorStore(editorStoreSelectors.setDraftId);
 
-    const exit = () => {
-        cleanUpEditor();
-
-        resetState();
-
-        navigate('/captures');
-    };
-
-    const discovers = {
-        done: (discoversSubscription: RealtimeSubscription) => {
+    const helpers = {
+        callFailed: (formState: any, subscription?: RealtimeSubscription) => {
+            const setFailureState = () => {
+                setFormState({
+                    status: CaptureCreationFormStatus.IDLE,
+                    exitWhenLogsClose: false,
+                    saveStatus: intl.formatMessage({
+                        id: 'captureCreation.status.failed',
+                    }),
+                    ...formState,
+                });
+            };
+            if (subscription) {
```

Comment:
This allows us to use the same function for both when an AJAX call totally dies (and we don't have a chance to start a subscription) and when it comes back but with an error.